Eastern-Research-Group/expert-query
# Feature/175 dependent filters

## Related Issues:
* [EQ-175](https://jira.epa.gov/browse/EQ-175)

## Main Changes:
* Added logic to send additional filter columns with the server requests for column values.
* Updated the client-side profile configuration to specify these filter dependencies. The configurations are all "one-way", so that  filters can still be added to an "upstream" field when a child option is selected.
* Updated the etl- and server-side profile configurations to create and access additional materialized views for these filtered values.
* Removed the logic to send an optional `context` field with domain values. This was only used in one or two cases, and now it would be used in fewer instances (e.g. when a column is being filtered by an additional column that wasn't included in the domain service). I think the occasional benefit isn't worth the added complexity.

## Steps To Test:
1. Run the etl with environment variables pointed at a local database.
2. Navigate to http://localhost:3000/attains/assessments.
3. Open the State drop-down input and note the available options.
4. Select a region from the Region select input.
5. The State drop-down input should be limited to the selected region (since the local database is being used, the exact options might differ).
6. Select a state from the State drop-down.
7. Confirm that the IDs listed in the Assessment Unit ID drop-down all belong to the selected state.
8. Check other configurations as specified in the `client/src/config/profiles.ts` file. These were the groupings discussed at a recent meeting with the EQ team, but more may be added in the future.
9. Confirm that the Organization ID and Location Text (in the Assessment Units profile) fields are still properly filtered by their "Source Select" filter inputs.
